spot_rl_experiments/spot_rl/envs/skill_manager.py

```diff
@@ -216,6 +234,109 @@ def nav(self, x: float, y: float, theta=float) -> Tuple[bool, str]:  # noqa
+        save_cone_search_images: bool = True,
+        pull_back: bool = True,
+    ) -> bool:
+        """
+        Perform Heuristic mobile navigation to the not very accurate pick target obtained from Aria glasses(x,y,theta)
+        Step 1 goto x,y,theta with head first navigation method
+        Step 2 search object_target using object detector in the list of -90 to 90 degrees (180 degrees) with 20 degrees interval
+        Step 3 If found update given x, y, theta with new
+        Step 4 Navigate to new x, y, theta in 50 steps
+        Step 5 turn off head first navigation
+        Step 6 Return Flag: bool signifying whether we found the object_target & whether is ready to pick ?
+
+        Args:
+            x (float): x coordinate of the nav target (in meters) specified in the world frame
+            y (float): y coordinate of the nav target (in meters) specified in the world frame
+            theta (float): yaw for the nav target (in radians) specified in the world frame
+            object_target: str object to search
+            image_search : spot_rl.utils.heuristic_nav.ImageSearch, Optional, default=None, ImageSearch (object detector wrapper), if none creates a new one for you uses OwlVit
+            save_cone_search_images: bool, optional, default= True, saves image with detections in each search cone
+            pull_back : bool, optional, default=True, pulls back x,y along theta direction
+        Returns:
+            bool: True if navigation was successful, False otherwise, if True you are good to fire .pick method
+
+        """
+        print(f"Original Nav targets {x, y, theta}")
+        if save_cone_search_images:
+            previously_saved_images = glob("imagesearch*.png")
+            for f in previously_saved_images:
+                os.remove(f)
+
+        if image_search is None:
+            image_search = ImageSearch(
+                corner_static_offset=0.5, use_yolov8=False, visualize=True
+            )
+        self.nav_controller.nav_env.enable_nav_goal_change()
+        (x, y) = (
+            pull_back_point_along_theta_by_offset(x, y, theta, 0.2)
+            if pull_back
+            else (x, y)
+        )
+        print(
+            f"Nav targets adjusted on the theta direction ray {x, y, np.degrees(theta)}"
+        )
+        backup_steps = self.nav_controller.nav_env.max_episode_steps
+        self.nav_controller.nav_env.max_episode_steps = 50
+        self.nav(x, y, theta)
+        self.nav_controller.nav_env.max_episode_steps = backup_steps
+        spot: Spot = self.spot
+        spot.open_gripper()
+        gaze_arm_angles = deepcopy(self.pick_config.GAZE_ARM_JOINT_ANGLES)
+        spot.set_arm_joint_positions(np.deg2rad(gaze_arm_angles), 1)
+        time.sleep(1.2)
+        found, (x, y, theta), visulize_img = image_search.search(
+            object_target, *get_me_arguments_for_image_search_fn(spot, 0)
+        )
+        rate = 20
+        if not found:
```

Comment:
There is a duplicated code here. What is the difference between nav_mobile_hueristic and aria_XXXX code?

Reply:
same as above, forgot to remove from heuristic_nav.py